### PR TITLE
PS-7694 [8.0]: Fix compilation issues with gcc-11

### PR DESCRIPTION
**Pull request by inikep**
Fix the following 8.0 issues:
```
In file included from /home/vsts/work/1/s/include/my_alloc.h:43,
                 from /home/vsts/work/1/s/mysys/array.cc:38:
/home/vsts/work/1/s/include/my_alloc.h: In member function ‘void* MEM_ROOT::Alloc_aligned(size_t, size_t)’:
/home/vsts/work/1/s/include/my_alloc.h:161:40: error: ‘uintptr_t’ in namespace ‘std’ does not name a type
  161 |         MY_ALIGN(reinterpret_cast<std::uintptr_t>(ptr), alignment));
      |                                        ^~~~~~~~~
/home/vsts/work/1/s/include/my_pointer_arithmetic.h:34:27: note: in definition of macro ‘MY_ALIGN’
   34 | #define MY_ALIGN(A, L) (((A) + (L)-1) & ~((L)-1))

/data/mysql-server/percona-8.0/router/src/harness/tests/test_net_ts_impl_resolver.cc: In member function ‘virtual void NetTS_impl_resolver_getnameinfo_invalid_sockaddr_size_Test::TestBody()’:
/data/mysql-server/percona-8.0/router/src/harness/tests/test_net_ts_impl_resolver.cc:114:55: error: ‘serv’ may be used uninitialized [-Werror=maybe-uninitialized]
  114 |       name.data(), name.size(), serv.data(), serv.size(),
      |                                              ~~~~~~~~~^~

/data/mysql-server/percona-8.0/sql-common/sql_string.cc: In member function ‘bool String::mem_realloc(size_t, bool)’:
/data/mysql-server/percona-8.0/sql-common/sql_string.cc:111:20: error: ‘numeric_limits’ is not a member of ‘std’
  111 |     if (len > std::numeric_limits<uint32>::max()) return true;

/data/mysql-server/percona-8.0/router/src/harness/tests/plugins/lifecycle.cc: In function ‘void lifecycle_init(int)’:
/data/mysql-server/percona-8.0/router/src/harness/tests/plugins/lifecycle.cc:452:29: error: loop variable ‘key’ of type ‘const string&’ {aka ‘const std::__cxx11::basic_string<char>&’} binds to a temporary constructed from type ‘const char* const’ [-Werror=range-loop-construct]
  452 |     for (const std::string &key : {"instance1", "instance2", "instance3"}) {

/data/mysql-server/percona-8.0/sql/server_component/mysql_query_attributes_imp.cc: In function ‘String* query_parameter_val_str(const PS_PARAM*, const CHARSET_INFO*)’:
/data/mysql-server/percona-8.0/sql/server_component/mysql_query_attributes_imp.cc:277:16: error: ‘static void String::operator delete(void*, size_t)’ called on pointer returned from a mismatched allocation function [-Werror=mismatched-new-delete]

In file included from /data/mysql-server/percona-8.0/components/mysqlbackup/backup_page_tracker.h:29,
                 from /data/mysql-server/percona-8.0/components/mysqlbackup/backup_page_tracker.cc:26:
/data/mysql-server/percona-8.0/include/mysql/components/services/page_track_service.h:58:36: error: ‘size_t’ has not been declared
   58 |                                    size_t buf_len, int num_pages,

/data/mysql-server/percona-8.0/sql/sql_cursor.cc: In member function ‘virtual bool Query_result_materialize::prepare(THD*, const mem_root_deque<Item*>&, SELECT_LEX_UNIT*)’:
/data/mysql-server/percona-8.0/sql/sql_cursor.cc:469:71: error: ‘static void Server_side_cursor::operator delete(void*, MEM_ROOT*, const std::nothrow_t&)’ called on pointer returned from a mismatched allocation function [-Werror=mismatched-new-delete]
  469 |   materialized_cursor = new (thd->mem_root) Materialized_cursor(result);

/data/mysql-server/percona-8.0/sql/sql_prepare.cc: In member function ‘virtual bool Protocol_local::send_eof(uint, uint)’:
/data/mysql-server/percona-8.0/sql/sql_prepare.cc:3886:67: error: ‘static void Ed_result_set::operator delete(void*, MEM_ROOT*, const std::nothrow_t&)’ called on pointer returned from a mismatched allocation function [-Werror=mismatched-new-delete]
 3886 |       Ed_result_set(m_rset, m_fields, m_column_count, &m_rset_root);
      |                                                                   ^
/data/mysql-server/percona-8.0/sql/sql_prepare.cc:3886:67: note: returned from ‘void* operator new(size_t, MEM_ROOT*, const std::nothrow_t&)’

/data/mysql-server/percona-8.0/storage/innobase/os/os0file.cc: In function ‘dberr_t os_file_copy_read_write(os_file_t, os_offset_t, os_file_t, os_offset_t, uint)’:
/data/mysql-server/percona-8.0/storage/innobase/os/os0file.cc:6153:41: error: ‘buf’ may be used uninitialized [-Werror=maybe-uninitialized]
 6153 |   buf_ptr = static_cast<char *>(ut_align(buf, UNIV_SECTOR_SIZE));
      |                                 ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from /data/mysql-server/percona-8.0/storage/innobase/include/ut0byte.h:97,
                 from /data/mysql-server/percona-8.0/storage/innobase/include/ut0new.h:149,
                 from /data/mysql-server/percona-8.0/storage/innobase/include/sync0types.h:42,
                 from /data/mysql-server/percona-8.0/storage/innobase/include/univ.i:593,
                 from /data/mysql-server/percona-8.0/storage/innobase/os/file.h:47,
                 from /data/mysql-server/percona-8.0/storage/innobase/include/os0file.h:47,
                 from /data/mysql-server/percona-8.0/storage/innobase/os/os0file.cc:41:
/data/mysql-server/percona-8.0/storage/innobase/include/ut0byte.ic:75:7: note: by argument 1 of type ‘const void*’ to ‘void* ut_align(const void*, ulint)’ declared here
   75 | void *ut_align(const void *ptr, /*!< in: pointer */
      |       ^~~~~~~~
/data/mysql-server/percona-8.0/storage/innobase/os/os0file.cc:6150:8: note: ‘buf’ declared here
 6150 |   char buf[BUF_SIZE + UNIV_SECTOR_SIZE];
 ```

### CODE REVIEW

File: components/mysqlbackup/backup_page_tracker.h [hunk 1]
```diff
@@ -28,6 +28,7 @@
 
 #include <mysql/components/services/page_track_service.h>
 
+#include <stddef.h>  // size_t
```
Comment (percona-ysorokin) [inline components/mysqlbackup/backup_page_tracker.h:31]: `<cstddef>` is more c++-ish

File: include/mysql/components/services/page_track_service.h [hunk 2]
```diff
@@ -27,6 +27,7 @@
 #include <mysql/components/service.h>
 #include <functional>
 
+#include <stddef.h>  // size_t
```
Comment (percona-ysorokin) [inline include/mysql/components/services/page_track_service.h:30]: `<cstddef>` is more c++-ish

**Review by percona-ysorokin — APPROVED** [review]
LGTM with small comment that can be ignored

**Comment by inikep** [comment]
Tested at https://ps80.cd.percona.com/job/percona-server-8.0-param/927/

**Pull request merged**